witness/golang/omniwitness/usbarmory/main.go

```diff
@@ -72,9 +120,32 @@ func main() {
 		WitnessSigner:   signer,
 		WitnessVerifier: verifier,
 	}
-	// TODO(mhutchinson): replace this with a real persistence layer.
-	p := inmemory.NewPersistence()
+	p := storage.NewSlotPersistence(part)
+	if err := p.Init(); err != nil {
+		glog.Exitf("Failed to create persistence layer: %v", err)
+	}
 	if err := omniwitness.Main(ctx, opConfig, p, httpListener, httpClient); err != nil {
 		glog.Exitf("Main failed: %v", err)
 	}
 }
+
+func openStorage() *slots.Partition {
+	// dev is our access to the MMC storage.
+	dev := &storage.Device{
+		Card: usbarmory.MMC,
+	}
+	bs := uint(usbarmory.MMC.Info().BlockSize)
+	geo := slots.Geometry{
+		Start:  slotsPartitionOffsetBytes / bs,
+		Length: slotsPartitionLengthBytes / bs,
+	}
+	for i, sl := uint(0), slotSizeBytes/bs; i < geo.Length; i += sl {
```

Comment:
I'd pull out `sl` to above here. In my experience it's strongly implied that the variables in the initialization block of a `for` loop are going to be modified during the loop. This one isn't, and seems to be a constant.

Reply:
Sure, done.